modules/apps/server/server-admin-web/src/main/java/com/liferay/server/admin/web/internal/portlet/action/EditServerMVCActionCommand.java

```diff
@@ -220,11 +233,34 @@ else if (cmd.equals("verifyPluginTables")) {
 		sendRedirect(actionRequest, actionResponse, redirect);
 	}
 
-	private void _addLogLevel(ActionRequest actionRequest) throws Exception {
-		String loggerName = ParamUtil.getString(actionRequest, "loggerName");
-		String priority = ParamUtil.getString(actionRequest, "priority");
+	@Override
+	public String getOSGiServiceIdentifier() {
+		return EditServerMVCActionCommand.class.getName();
+	}
+
+	private static void _resetLogLevels(
+		Map<String, String> logLevels, Map<String, String> customLogSettings) {
+
+		for (Map.Entry<String, String> logLevel : logLevels.entrySet()) {
+			Log4JUtil.setLevel(
+				logLevel.getKey(), logLevel.getValue(),
+				customLogSettings.containsKey(logLevel.getKey()));
+		}
+	}
+
+	private static void _updateLogLevels(
+		Map<String, String> logLevels, String osgiServiceIdentifier) {
```

Comment:
@tinatian there is no need to have this "String osgiServiceIdentifier" as method parameter, right? On the other side of the cluster receiver, it can just use its own class name to lookup local instance of this EditServerMVCActionCommand.

Reply:
@shuyangzhou You can only do that when getOSGiServiceIdentifier is returning the class name as osgi service identifier, even though this is what we always do, but it is actually a hidden rule, thus I still prefer to pass in the parameter to make it more readable rather than making people figure out this hidden rule.

```
public String getOSGiServiceIdentifier() {
		return EditServerMVCActionCommand.class.getName();
	}
```